File: react/reactproject/src/models/turn.ts
export class Turn {
    TurnId: number;
    TurnUserId: string;
    DoctorName: string;
    DoctorCode: number;
    DateTurn: Date;
    TurnLocate: string;
    Hour: string;
    ArrivalConfirmation: boolean;

    constructor(data: Partial<Turn> = {}) {
        this.TurnId = data.TurnId ?? 0;
        this.TurnUserId = data.TurnUserId ?? "Unidentified user";
        this.DoctorName = data.DoctorName ?? "Unidentified doctor";
        this.DoctorCode = data.DoctorCode ?? 0;
        this.DateTurn = data.DateTurn ?? new Date();
        this.TurnLocate = data.TurnLocate ?? "Unidentified Location";
        this.ArrivalConfirmation = data.ArrivalConfirmation ?? false;

        // חישוב שעה בפורמט 12:30
        const hours = this.DateTurn.getHours().toString().padStart(2, '0');
        const minutes = this.DateTurn.getMinutes().toString().padStart(2, '0');
        this.Hour = `${hours}:${minutes}`;
    }
}

File: Net/FinalProject/FinalProject.Clean/Models/Turn.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinalProject.Core.Models
{
    public class Turn
    {
        [Key]
        public int TurnId { get; set; }
        public int TurnUserId { get; set; }
        public string DoctorName { get; set; }
        public int DoctorCode { get; set; }
        public DateTime DateTurn { get; set; }
        public string TurnLocate { get; set; }
        public string Hour { get; set; }
        public bool ArrivalConfirmation { get; set; }
        public User User { get; set; }
    }
}
